src/frontend/core/executor/impl/TriangleCountExecutor.cpp

```diff
@@ -0,0 +1,1560 @@
+/**
+Copyright 2019 JasminGraph Team
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+ */
+
+#include "TriangleCountExecutor.h"
+
+using namespace std::chrono;
+
+Logger triangleCount_logger;
+std::vector<std::vector<string>> TriangleCountExecutor::fileCombinations;
+std::map<std::string, std::string> TriangleCountExecutor::combinationWorkerMap;
+std::map<long, std::map<long, std::vector<long>>> TriangleCountExecutor::triangleTree;
+std::mutex processStatusMutex;
+std::mutex fileCombinationMutex;
+std::mutex triangleTreeMutex;
+
+TriangleCountExecutor::TriangleCountExecutor() {
+
+}
+
+TriangleCountExecutor::TriangleCountExecutor(SQLiteDBInterface db, PerformanceSQLiteDBInterface perfDb, JobRequest jobRequest) {
+    this->sqlite = db;
+    this->perfDB = perfDb;
+    this->request = jobRequest;
+}
+
+void TriangleCountExecutor::execute() {
+    std::string masterIP= request.getParameter(Conts::PARAM_KEYS::MASTER_IP);
+    std::string graphId = request.getParameter(Conts::PARAM_KEYS::GRAPH_ID);
+    std::string canCalibrate = request.getParameter(Conts::PARAM_KEYS::CAN_CALIBRATE);
```

Comment:
Can't we change the type of the variable `canCalibrate` to boolean? That will remove the requirement of string equality check in the latter half of this code.

Reply:
Changed parameter to boolean